bundles/binding/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/EcobeeBinding.java

```diff
@@ -0,0 +1,1023 @@
+	protected static final String CONFIG_SCOPE = "scope";
+	protected static final String CONFIG_TEMP_SCALE = "tempscale";
+
+	static {
+		// Register bean type converters
+		ConvertUtils.register(new Converter() {
+
+			@SuppressWarnings("rawtypes")
+			@Override
+			public Object convert(Class type, Object value) {
+				if (value instanceof DecimalType) {
+					return Temperature.fromLocalTemperature(((DecimalType) value).toBigDecimal());
+				} else {
+					return null;
+				}
+			}
+		}, Temperature.class);
+		ConvertUtils.register(new Converter() {
+
+			@SuppressWarnings("rawtypes")
+			@Override
+			public Object convert(Class type, Object value) {
+				if (value instanceof StringType) {
+					return HvacMode.forValue(value.toString());
+				} else {
+					return null;
+				}
+			}
+		}, HvacMode.class);
+		ConvertUtils.register(new Converter() {
+
+			@SuppressWarnings("rawtypes")
+			@Override
+			public Object convert(Class type, Object value) {
+				if (value instanceof DecimalType) {
+					return ((DecimalType) value).intValue();
+				} else {
+					return null;
+				}
+			}
+		}, Integer.class);
+		ConvertUtils.register(new Converter() {
+
+			@SuppressWarnings("rawtypes")
+			@Override
+			public Object convert(Class type, Object value) {
+				if (value instanceof StringType) {
+					return VentilatorMode.forValue(value.toString());
+				} else {
+					return null;
+				}
+			}
+		}, VentilatorMode.class);
+		ConvertUtils.register(new Converter() {
+
+			@SuppressWarnings("rawtypes")
+			@Override
+			public Object convert(Class type, Object value) {
+				if (value instanceof OnOffType) {
+					return ((OnOffType) value) == OnOffType.ON;
+				} else {
+					return null;
+				}
+			}
+		}, Boolean.class);
+		ConvertUtils.register(new Converter() {
+
+			@SuppressWarnings("rawtypes")
+			@Override
+			public Object convert(Class type, Object value) {
+				return value.toString();
+			}
+		}, String.class);
+		// FIXME: complete type conversions from states to bean property types
+	}
+
+	/**
+	 * the refresh interval which is used to poll values from the Ecobee server (optional, defaults to 60000ms)
+	 */
+	private long refreshInterval = 60000;
+
+	/**
+	 * A map of userids from the openhab.cfg file to OAuth credentials used to communicate with each app instance.
+	 */
+	private Map<String, OAuthCredentials> credentialsCache = new HashMap<String, OAuthCredentials>();
+
+	/**
+	 * used to store events that we have sent ourselves; we need to remember them for not reacting to them
+	 */
+	private Set<String> ignoreEventSet = new HashSet<String>();
+
+	/**
+	 * The most recently received list of revisions, or an empty Map if none have been retrieved yet.
+	 */
+	private Map<String, Revision> lastRevisionMap = new HashMap<String, Revision>();
+
+	public EcobeeBinding() {
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void activate() {
+		super.activate();
+		setProperlyConfigured(true);
```

Comment:
better don't call the setter here since the configuration (from updated()) must not be called already